ext/include/opentelemetry/ext/http/client/curl/http_client_curl.h

```diff
@@ -153,6 +153,33 @@ class Session : public http_client::Session
     });
   }
 
+  virtual std::unique_ptr<http_client::Response> SendRequestSync(
+      http_client::SessionState &session_state) noexcept override
+  {
+    is_session_active_ = true;
+    std::string url    = host_ + std::string(http_request_->uri_);
+    curl_operation_.reset(new HttpOperation(http_request_->method_, url, nullptr, RequestMode::Sync,
```

Comment:
Could we set `RequestMode::Sync` in `SendSync()` or just deduce it from `SendSync()` and `SendAsync()`. Is the inconsistency between ctor and `SendSync()` handled?

Reply:
We do need to know the actual sync mode in ctor. There are some initializations done in ctor, and the `sync mode` information is used to propagate the error/results to clients. 

>Is the inconsistency between ctor and SendSync() handled?

What kind of inconsistency we are talking about.